File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayOptTimestampValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayOptTimestampInValue.OptArrayOptTimestampValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayOptTimestampValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayOptTimestampValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.joda.time.DateTime[].class,
                OptArrayOptTimestampValue.class.getDeclaredField("optArrayOptTimestamp").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.joda.time.DateTime[].class,
                OptArrayOptTimestampValue.class.getMethod("getOptArrayOptTimestamp").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayOptTimestampValue.class,
                OptArrayOptTimestampValue.class.getMethod("setOptArrayOptTimestamp", org.joda.time.DateTime[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayOptTimestampValue().getOptArrayOptTimestamp());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayOptTimestampValue.scala
package com.dslplatform.ocd.values

import OptArrayOptTimestampInValue.OptArrayOptTimestampValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayOptTimestampValue extends SpecExtensions {
  def `Test property field type: Option[Array[Option[org.joda.time.DateTime]]]`() =
    checkType(
      typeOf[Option[Array[Option[org.joda.time.DateTime]]]]
    , typeOf[OptArrayOptTimestampValue].member("optArrayOptTimestamp": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayOptTimestampValue().optArrayOptTimestamp === None)
  }
}
